x-pack/plugins/actions/server/builtin_action_types/lib/connector_token_client.ts

```diff
@@ -219,6 +214,24 @@ export class ConnectorTokenClient {
       );
       return { hasErrors: true, connectorToken: null };
     }
+
+    if (isNaN(Date.parse(connectorTokensResult[0].attributes.expiresAt))) {
```

Comment:
If `Date.parse(expiresAt)` is `NaN`, then checking whether the token had expired would always return `false`, causing us to keep using the same access token repeatedly.